src/app.rs

```diff
@@ -0,0 +1,370 @@
+                                    KeyCode::Enter => {
+                                        let entr_txt: String = self.submit();
+                                        input_tx.send(format!("{}\r\n", entr_txt.clone())).unwrap();
+                                        if entr_txt.to_uppercase() == "EXIT" {
+                                            break;
+                                        }
+                                    }
+                                    KeyCode::Char('c')
+                                    if key.modifiers == KeyModifiers::from_name("CONTROL").unwrap() =>
+                                        {
+                                            if input_tx.send("stop\n".to_string()).is_err() {
+                                                self.output.push("Couldn't stop!".to_string());
+                                            }
+                                            if spam_handler.interrupted() {
+                                                res = input_tx.send("EXIT".to_string()).map_err(|e| io::Error::new(ErrorKind::Other, e.0));
+                                                break;
+                                            }
+                                        }
+                                    KeyCode::Char(c) => self.put_char(c),
+                                    KeyCode::Backspace => self.delete_char(),
+                                    KeyCode::Up => {
+                                        self.input = self.cmd_history.prev_cmd();
+                                        self.cursor_pos = self.input.len();
+                                    }
+                                    KeyCode::Down => {
+                                        self.input = self.cmd_history.next_cmd();
+                                        self.cursor_pos = self.input.len();
+                                    }
+                                    KeyCode::Left => self.cursor_left(),
+                                    KeyCode::Right => self.cursor_right(),
+                                    KeyCode::PageUp => self.scroll_up(),
+                                    KeyCode::PageDown => self.scroll_down(),
+                                    KeyCode::Esc => self.input_mode = InputMode::Normal,
+
+                                    _ => (),
+                                }
+                            }
+                            InputMode::Normal => {
+                                match key.code {
+                                    KeyCode::Up | KeyCode::PageUp => self.scroll_up(),
+                                    KeyCode::Down | KeyCode::PageDown => self.scroll_down(),
+                                    KeyCode::Esc => self.input_mode = InputMode::Insert,
+                                    _ => ()
+                                }
+                            }
+                        }
+                    }
+                }
+            }
```

Comment:
Could improve readability if we reduce the nesting here somehow

Reply:
Agreed. I combined some conditions from nested if statements and moved event handling into a separate function to help with this.